rcgen/tests/generic.rs

```diff
@@ -400,7 +400,29 @@ mod test_csr {
 		let csrp = CertificateSigningRequestParams::from_der(csr.der()).unwrap();
 
 		// Assert that our parsed parameters match our initial parameters
+		// Certificate parameters are #[non_exhaustive] and don't implement equality
```

Comment:
Do you have a hypothesis as to why we don't/can't implement `PartialEq`? Seems like we could and maybe should...

Reply:
I can't think of a reason not implement `PartialEq`. Same with `Eq`—I can't see a reason why we wouldn't want reflexive equality for certificate params. I've added a commit that derives both. I also threw in `Debug`, which we need for `assert_eq` and might be useful anyways.

Reply:
I can take any of these derives out if anyone feels strongly, of course.

Reply:
LGTM, thanks!